### PR TITLE
fix: Prod x feature deployment

### PR DESCRIPTION
**Pull request by PritishBudhiraja**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Add feature flag to point dashboard to custom pod
image

## How did you test it?
custom header should be passed in the request headers

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by Riddhiagrawal001 — DISMISSED** [review]

File: src/hooks/AuthHooks.res [hunk 1]
```diff
@@ -10,26 +10,44 @@ type sessionStorage = {
 external dictToObj: Js.Dict.t<'a> => {..} = "%identity"
 @val external atob: string => string = "atob"
 
-let getHeaders = (~uri, ~headers, ()) => {
+let getHeaders = (~uri, ~headers, ~xFeatureRoute, ()) => {
   let hyperSwitchToken = LocalStorage.getItem("login")->Js.Nullable.toOption
   let isMixpanel = uri->String.includes("mixpanel")
 
+  let headersForXFeature = (~token) =>
```
Comment (jainlokesh318) [inline src/hooks/AuthHooks.res:17]: This function can be outside

File: src/hooks/AuthHooks.res [hunk 2]
```diff
@@ -10,26 +10,44 @@ type sessionStorage = {
...
+    if (
+      uri->Js.String2.includes("lottie-files") || uri->Js.String2.includes("config/merchant-access")
+    ) {
+      headers->Dict.set("Content-Type", `application/json`)
+      headers->Dict.set("authorization", `Bearer ${token}`)
+      headers->Dict.set("api-key", `hyperswitch`)
+      headers
+    } else {
+      headers->Dict.set("authorization", `Bearer ${token}`)
```
Comment (jainlokesh318) [inline src/hooks/AuthHooks.res:26]: The common setters to the headers should not be repeated

File: src/hooks/AuthHooks.res [hunk 3]
```diff
@@ -10,26 +10,44 @@ type sessionStorage = {
...
+      xFeatureRoute
+        ? headersForXFeature(~token)
+        : {
+            headers->Dict.set("authorization", `Bearer ${token}`)
+            headers->Dict.set("api-key", `hyperswitch`)
+            headers
+          }
 
-    | None => headers
+    | None => xFeatureRoute ? headersForXFeature(~token="") : headers
```
Comment (jainlokesh318) [inline src/hooks/AuthHooks.res:48]: for xFeatureRoute the settig can be done first, then the token thing. Repetition will be reduced.

File: src/hooks/AuthHooks.res (hunk 2 above)
Comment (PritishBudhiraja) [reply src/hooks/AuthHooks.res]: updated

File: src/hooks/AuthHooks.res (hunk 1 above)
Comment (PritishBudhiraja) [reply src/hooks/AuthHooks.res]: updated

**Review by jainlokesh318 — APPROVED** [review]

**Review by JeevaRamu0104 — APPROVED** [review]

**Review by sagarnaikjuspay — APPROVED** [review]

**Review by Riddhiagrawal001 — APPROVED** [review]

**Pull request merged**